### PR TITLE
#167646812 create nav bar and app layout

### PR DESCRIPTION
**Pull request by timi-codes**
#### What does this PR do?
- create app header
- Configure sass to work with web-pack and jest
- Implemented the modularised project structure for store
#### Link to Pivotal Story
[#167646812](https://www.pivotaltracker.com/story/show/167646812)

### CODE REVIEW

**Review by Onyimatics — APPROVED** [review]
This looks good to me.

**Review by codexempire — APPROVED** [review]
This looks good to me.

File: __test__/components/navbar.spec.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      <Router>
+        <NavBar isLoggedIn={false} />
+      </Router>
+    );
+    wrapper.find('#hamburger').simulate('click');
+    mockState(true);
+    expect(wrapper.find('#hamburger').hasClass('active')).toEqual(true);
+  });
+
+  it('should test when user is logged in', () => {
```
Comment (segunolalive) [inline __test__/components/navbar.spec.js]: ```it displays something when the *isLoggedIn* prop is set to true```

Comment (segunolalive) [reply __test__/components/navbar.spec.js]: Your tests should be about what the UI displays based on application state.

So, in this case, it should test that the avatar is rendered when `isLoggedin = true`, but not rendered when set to `false`. It also renders the `AuthButtons` when set to `false` but not when set to `true`

File: __test__/components/navbar.spec.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    );
+    expect(
+      wrapper
+        .find('#menu')
+        .childAt(0)
+        .hasClass('avatar')
+    ).toEqual(true);
+  });
+
+  it('should display auth button when user is not logged in ', () => {
```
Comment (segunolalive) [inline __test__/components/navbar.spec.js]: it should display AuthButtons when `isLoggedIn` is set to false.

File: __test__/components/navbar.spec.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    expect(
+      wrapper
+        .setProps({
+          children: <AuthButtons id="auth-buttons" show />
+        })
+        .find('#auth-buttons')
+        .children()
+    ).toHaveLength(2);
+
+    expect(
```
Comment (segunolalive) [inline __test__/components/navbar.spec.js:66]: I like this 😍

Comment (segunolalive) [reply __test__/components/navbar.spec.js:66]: This tested what was displayed based on the props.

File: __test__/reducers/auth.spec.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import auth from '@modules/auth/reducer';
+
+describe('Test auth reducer function', () => {
```
Comment (segunolalive) [inline __test__/reducers/auth.spec.js]: `describe(Auth Reducer ...`

Comment (segunolalive) [reply __test__/reducers/auth.spec.js]: You can remove these

File: src/components/NavBar/index.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        <Link to="/" title="Error Swag Logo" className="flex items-center">
+          <img
+            src={logo}
+            alt="Errorswag logo"
+            className="sm:w-8/12 md:w-full"
+          />
+        </Link>
+      </div>
+      <div className="flex">
+        <Link to="/search" className="small-button" title="Search">
```
Comment (segunolalive) [inline src/components/NavBar/index.js]: You can use any of these.
aria-labelledby
aria-label
aria-describedby (for long description. announced afterwards)
title

Title is actually the least supported. The others are preferred over title by screen readers

File: src/components/NavBar/index.js [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+      }`}
+      type="button"
+      aria-expanded={menuOpen}
+      aria-label="Menu"
+    >
+      <span />
+    </button>
+  );
+};
+
```
Comment (segunolalive) [inline src/components/NavBar/index.js:86]:
```js
const MenuIcon = ({ isLoggedIn, toggleMenu, menuOpen }) => {
  if (isLoggedIn) {
    return <button
      id="avatar"
      className="avatar flex items-center mx-2 focus:outline-none"
      type="button"
      aria-label="Menu"
    >
      <AvatarIcon className="w-8 h-8" />
    </button>
  )}
  return (
    <button
      id="hamburger"
      onClick={toggleMenu}
      className={`hamburger small-button sm:block md:hidden${
        menuOpen ? ' active' : ''
      }`}
      type="button"
      aria-expanded={menuOpen}
      aria-label="Menu"
    >
      <span />
    </button>
  );
};
```

File: src/components/NavBar/index.js [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+      type="button"
+      aria-label="Menu"
+    >
+      <AvatarIcon className="w-8 h-8" />
+    </button>
+  ) : (
+    <button
+      id="hamburger"
+      onClick={toggleMenu}
+      className={`hamburger small-button sm:block md:hidden${
```
Comment (segunolalive) [inline src/components/NavBar/index.js]:
```js
<button data-active={menuOpen} />
```
```css

button[data-active] {
  ...
}

```

File: src/main.scss [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+@tailwind utilities;
+
+@import '../public/styles/colors.scss';
+@import url('https://fonts.googleapis.com/css?family=Poppins:400,500,600,700,800,900|Roboto&display=swap');
+
+:root {
+  --font: 'Poppins', sans-serif;
+  font-family: var(--font);
+  font-style: normal;
+}
```
Comment (segunolalive) [inline src/main.scss:16]: set the default text color here as well

**Review by segunolalive — CHANGES_REQUESTED** [review]
Looks good in general. Make the changes required.

File: __test__/App.spec.js [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+    const wrapper = mount(
+      <BrowserRouter>
+        <HomePage />
+      </BrowserRouter>
+    );
+
+    expect(toJson(wrapper)).toMatchSnapshot();
+  });
+
+  it('should work fine Home Page', () => {
```
Comment (olorunwalawrence) [inline __test__/App.spec.js]: please kindly change the text of this it block to describe what the test does.
the snapshot shows NotFoundPage. kindly look into that.

File: src/pages/notFoundPage.js [hunk 10]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import logo from '../assets/images/notfound.png';
+
+const NotFoundPage = () => (
+  <div className="flex items-center justify-center flex-col h-screen">
+    <img src={logo} alt="404 Page" className="sm:w-2/4 md:w-1/5" />
+    <h2 className="text-secondary my-6 md:w-2/5 text-center text-sm font-semibold">
+      THE PAGE YOU ARE LOOKING FOR MIGHT HAVE BEEN REMOVED HAD ITS NAME CHANGED
```
Comment (olorunwalawrence) [inline src/pages/notFoundPage.js]: THE PAGE YOU ARE LOOKING FOR MIGHT HAVE BEEN REMOVED, IT'S NAME CHANGED
      OR ITS TEMPORARILY UNAVAILABLE

please look into this and make the change

**Review by olorunwalawrence — CHANGES_REQUESTED** [review]
looks good, kindly attend to the few changes

File: src/pages/notFoundPage.js (hunk 10 above)
Comment (timi-codes) [reply src/pages/notFoundPage.js]: What changes should be made?

File: __test__/App.spec.js (hunk 9 above)
Comment (timi-codes) [reply __test__/App.spec.js]: Thanks for the feedback. This has been fixed

**Review by olorunwalawrence — APPROVED** [review]
This looks good. 👍

File: __test__/App.spec.js [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+    const wrapper = mount(comp);
+
+    expect(wrapper.find('Home')).toBeTruthy();
+  });
+  it('should not crash app', () => {
+    const wrapper = mount(<App />);
+
+    expect(toJson(wrapper)).toMatchSnapshot();
+  });
+  it('should work fine on Login Page', () => {
```
Comment (segunolalive) [inline __test__/App.spec.js:55]: please remove this

File: __test__/App.spec.js [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+    const wrapper = mount(
+      <BrowserRouter>
+        <LoginPage />
+      </BrowserRouter>
+    );
+
+    expect(toJson(wrapper)).toMatchSnapshot();
+  });
+  it('it should render Home Page without crashing', () => {
```
Comment (segunolalive) [inline __test__/App.spec.js:64]: please remove this.

File: __test__/App.spec.js [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+describe('Application test', () => {
+  let store;
+  beforeEach(() => {
+    store = createStore(
+      combineReducers({
+        auth
+      })
+    );
+  });
+  it('should render index page', () => {
```
Comment (segunolalive) [inline __test__/App.spec.js:36]: This test doesn't comprehensively test the app component

**Pull request merged**